MyApplication/app/src/main/java/net/serkanbal/myapplication/MainActivity.java

```diff
@@ -0,0 +1,139 @@
+package net.serkanbal.myapplication;
+
+import android.app.job.JobInfo;
+import android.app.job.JobScheduler;
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Bundle;
+import android.support.v4.content.LocalBroadcastManager;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.TextView;
+
+public class MainActivity extends AppCompatActivity {
+    TextView mJob1Before, mJob1After, mJob2Before, mJob2After, mJob3Before, mJob3After;
+    BroadcastReceiver mReceiver, mReceiver2, mReceiver3;
+    Integer mFirstNumber = 0, mSecondNumber = 0;
+    String mFirstDate = "", mSecondDate = "", mFirstToken = "", mSecondToken = "";
+    public static final int FIRST_JOB_ID = 1;
+    public static final int SECOND_JOB_ID = 2;
+    public static final int THIRD_JOB_ID = 3;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mJob1Before = (TextView) findViewById(R.id.job1_before);
+        mJob1After = (TextView) findViewById(R.id.job1_after);
+        mJob2Before = (TextView) findViewById(R.id.job2_before);
+        mJob2After = (TextView) findViewById(R.id.job2_after);
+        mJob3Before = (TextView) findViewById(R.id.job3_before);
+        mJob3After = (TextView) findViewById(R.id.job3_after);
```

Comment:
I suggest giving these more descriptive names, like `oldRandNumberView` or `newRandTokenView`, etc. That would make the code much more readable.